### DISCUSSION TITLE
Freedom to chose the output dir name and build dir name

### DISCUSSION THREAD

**Discussion by kasir-barati** [keycloakify/keycloakify#294] [Ideas]
As you can see here you hardcoded the out dirname and also input dirname: https://github.com/keycloakify/keycloakify/blob/main/src/bin/keycloakify/build-paths.ts#L7-L9

I would love to see the ability to pass them to the `npx keycloakify`. Maybe:
- `npx keycloakify --outdir ./keycloak/theme`
- `npx keycloakify --indir ./dist`
- Or inside the `package.json` next to other build options

And if developer did not pass them, It can uses the hardcoded default values.

**Comment by kasir-barati** [comment]
Why I am saying this? Because I do have a monorepo and wanted to send the final result to the root of the project next to where the terraform codes reside. TBH there is no difference where the outdir would be since I can copy it from anywhere to anywhere I want with Terraform and Docker :sweat_smile:  but anyway I just wanted to share why I came up with this idea.

**Comment by garronej** [comment]
Hi @kasir-barati,  
We try avoiding adding options unless they are completely neccessary for some usecase.  
I feel like it would just be a matter of `npx keycloakify && mv build_keycloak whatever` to adress this usecase wouldn't it?  

I think I remember this having been requested already also I can't find where...

**Reply by lordvlad** [reply]
> I think I remember this having been requested already also I can't find where...

https://github.com/keycloakify/keycloakify/issues/87

**Reply by garronej** [reply]
Oh yeah that's right thanks !

**Comment by willwill96** [comment]
I believe this support was added here:
https://github.com/keycloakify/keycloakify/pull/287

Although it is only currently customizable through the `package.json` (and not CLI args)

An example using the default options would be:

```
//package.json
{
  "keycloakify": {
    "reactAppBuildDirPath: "./build",
    "keycloakifyBuildDirPath: "./build_keycloak"
  }
}
```

I mainly added it to make testing easier, but I think it could be supported via CLI args if you want to put together a PR!

The value is currently being pulled from here: https://github.com/keycloakify/keycloakify/blob/4ece6457fde3a6c835a4e9adca9bd9624427d22c/src/bin/keycloakify/BuildOptions.ts#L25-L28  

EDIT from @garronej: I updated the answer to reflect the refactor I just did.
